arangod/Pregel/Worker.cpp

```diff
@@ -778,7 +785,12 @@ void Worker<V, E, M>::finalizeRecovery(VPackSlice const& data) {
 
 template <typename V, typename E, typename M>
 void Worker<V, E, M>::_callConductor(std::string const& path, VPackBuilder const& message) {
-  if (ServerState::instance()->isRunningInCluster() == false) {
+  application_features::ApplicationServer& server = _config.vocbase()->server();
+  if (server.isStopping()) {
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_SHUTTING_DOWN);
```

Comment:
Have you tried this? From my experience Pregel is not very exception safe... :disappointed:

Reply:
Yes, tried it.
I agree overall it currently does not respond nicely to errors, exception, server shutdown. We should improve this further (in a seperate PR).